startup/GafferScene/renderPassTypes.py

```diff
@@ -0,0 +1,520 @@
+	processor["__catcherAndCasterFilter"]["in"].setInput( processor["in"] )
+
+	processor["__filterQuery"] = GafferScene.FilterQuery()
+	processor["__filterQuery"]["scene"].setInput( processor["in"] )
+	processor["__filterQuery"]["filter"].setInput( processor["__catcherAndCasterFilter"]["casters"] )
+	processor["__filterQuery"]["location"].setValue( "/" )
+
+	processor["__allReflectionExcluded"] = GafferScene.CustomAttributes()
+	processor["__allReflectionExcluded"]["in"].setInput( processor["in"] )
+	processor["__allReflectionExcluded"]["attributes"].addChild( Gaffer.NameValuePlug( "", Gaffer.BoolPlug() ) )
+	processor["__allReflectionExcluded"]["attributes"][0]["name"].setInput( processor["__attributeSpreadsheet"]["out"]["name"] )
+	processor["__allReflectionExcluded"]["global"].setValue( True )
+
+	# __allReflectionExcluded is only required when reflection casters do not include the root of the scene.
+	processor["__allReflectionExcludedExpression"] = Gaffer.Expression()
+	processor["__allReflectionExcludedExpression"].setExpression(
+		"""parent["__allReflectionExcluded"]["enabled"] = not parent["__filterQuery"]["exactMatch"]"""
+	)
+
+	processor["__reflectionInclusions"] = GafferScene.AttributeTweaks()
+	processor["__reflectionInclusions"]["in"].setInput( processor["__allReflectionExcluded"]["out"] )
+	processor["__reflectionInclusions"]["tweaks"].addChild( Gaffer.TweakPlug( "", Gaffer.BoolPlug( defaultValue = True ), mode = Gaffer.TweakPlug.Mode.CreateIfMissing ) )
+	processor["__reflectionInclusions"]["tweaks"][0]["name"].setInput( processor["__attributeSpreadsheet"]["out"]["name"] )
+	processor["__reflectionInclusions"]["filter"].setInput( processor["__catcherAndCasterFilter"]["casters"] )
+
+	processor["__reflectionExclusions"] = GafferScene.AttributeTweaks()
+	processor["__reflectionExclusions"]["in"].setInput( processor["__reflectionInclusions"]["out"] )
+	processor["__reflectionExclusions"]["tweaks"].addChild( Gaffer.TweakPlug( "", Gaffer.BoolPlug( defaultValue = False ), mode = Gaffer.TweakPlug.Mode.CreateIfMissing ) )
+	processor["__reflectionExclusions"]["tweaks"][0]["name"].setInput( processor["__attributeSpreadsheet"]["out"]["name"] )
+	processor["__reflectionExclusions"]["filter"].setInput( processor["__catcherAndCasterFilter"]["catchers"] )
+
+	# Cycles doesn't support fallback values in its `attribute` shader so we provide a fallback attribute.
+	processor["__fallbackAttributes"] = GafferScene.AttributeTweaks()
+	processor["__fallbackAttributes"]["in"].setInput( processor["__reflectionExclusions"]["out"] )
+	processor["__fallbackAttributes"]["filter"].setInput( processor["__catcherAndCasterFilter"]["catchers"] )
+	processor["__fallbackAttributes"]["localise"].setValue( True )
+	processor["__fallbackAttributes"]["tweaks"].addChild( Gaffer.TweakPlug( "user:reflectionCatcher:roughness", Gaffer.FloatPlug( defaultValue = 0.15 ), mode = Gaffer.TweakPlug.Mode.CreateIfMissing ) )
```

Comment:
The `user:` namespace is kindof meant to be the province of the user, so we wouldn't give any special meaning to anything in there. I think ideally we'd use `renderPass:reflectionCatcher:roughness` or something like that, but then of course it won't end up as a queryable attribute in the renderer.

Is there any merit to using `renderPass:reflectionCatcher:roughness` publicly, but then shuffling to `user:__renderPass:reflectionCatcher:roughness` internally? Or does that just muddy the waters further?

Reply:
From discussion offline, doing the `renderPass:` to `user:` shuffle seems potentially confusing once we have adaptors in the Scene Inspector, but I've renamed these to `user:renderPass:...` so they remain queryable in the renderer but are also better namespaced so we could more easily shuffle `user:renderPass:` to `renderPass:` for compatibility at some point in the future... e3f876325148ac2e44a25ec7aa5a3c5dde21fa39

I also took the opportunity to test these attributes in 1f6e7db11ac483644eaf5ced7448276b038bf688, much simpler to add with the base RenderAdaptorTest!